From fe7504f824e0e59f182b3e2eaffa4514c7bdbae1 Mon Sep 17 00:00:00 2001
From: AndreyVV-100 <viazovtsev.av@phystech.edu>
Date: Fri, 23 Jul 2021 12:48:03 +0300
Subject: [PATCH 007/245] Add new files for running criu-coredump via python 2
 or 3

Previous commit added support for python3 in criu-coredump. For convenience,
add two files (coredump-python2 and coredump-python3) that start
criu-coredump with respective python version. Edit env.sh accordingly.

Signed-off-by: Andrey Vyazovtsev <viazovtsev.av@phystech.edu>
---
 coredump/coredump-python2               | 6 ++++++
 coredump/coredump-python3               | 6 ++++++
 coredump/{criu-coredump => coredump.py} | 1 -
 test/others/env.sh                      | 2 +-
 4 files changed, 13 insertions(+), 2 deletions(-)
 create mode 100755 coredump/coredump-python2
 create mode 100755 coredump/coredump-python3
 rename coredump/{criu-coredump => coredump.py} (97%)

diff --git a/coredump/coredump-python2 b/coredump/coredump-python2
new file mode 100755
index 000000000..3a15c90a3
--- /dev/null
+++ b/coredump/coredump-python2
@@ -0,0 +1,6 @@
+#!/usr/bin/env python2
+
+import coredump
+
+if __name__ == '__main__':
+	coredump.main()
diff --git a/coredump/coredump-python3 b/coredump/coredump-python3
new file mode 100755
index 000000000..82ec6b855
--- /dev/null
+++ b/coredump/coredump-python3
@@ -0,0 +1,6 @@
+#!/usr/bin/env python3
+
+import coredump
+
+if __name__ == '__main__':
+	coredump.main()
diff --git a/coredump/criu-coredump b/coredump/coredump.py
similarity index 97%
rename from coredump/criu-coredump
rename to coredump/coredump.py
index d3113d372..56ba54083 100755
--- a/coredump/criu-coredump
+++ b/coredump/coredump.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 import argparse
 import os
 
diff --git a/test/others/env.sh b/test/others/env.sh
index b514e87d9..e2f63eee3 100755
--- a/test/others/env.sh
+++ b/test/others/env.sh
@@ -13,5 +13,5 @@ fi
 #export PYTHON
 CRIT=$(readlink -f `dirname ${BASH_SOURCE[0]}`/../../crit/crit-"${PYTHON}")
 crit=$CRIT
-CRIU_COREDUMP=$(readlink -f `dirname ${BASH_SOURCE[0]}`/../../criu-coredump/criu-coredump)
+CRIU_COREDUMP=$(readlink -f `dirname ${BASH_SOURCE[0]}`/../../criu-coredump/coredump-"${PYTHON}")
 criu_coredump=$CRIU_COREDUMP
-- 
2.35.1

